sdk/core/azure-core/src/main/java/com/azure/core/implementation/exception/UnexpectedLengthException.java

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.implementation.exception;
+
+public class UnexpectedLengthException extends IllegalStateException {
```

Comment:
You are making this class non-public API. Do you ever document anywhere in the storage APIs that this is thrown in certain situations?

Reply:
Good call. I will update the API which will throw the exception. This is not intended to use by client. It is for http request validation.